tests/plugins/colorbutton/colorbutton.js

```diff
@@ -164,6 +164,32 @@
 
 				wait();
 			} );
-		}
+		},
+
+		// (#1084)
+		'test changing text color to automatic': testAutomaticColor(),
+
+		// (#1084)
+		'test changing background color to automatic': testAutomaticColor( true )
 	} );
+
+	function testAutomaticColor( background ) {
+		return function() {
+			var editor = this.editor,
+				bot = this.editorBot,
+				colorBtn = editor.ui.get( background ? 'BGColor' : 'TextColor' );
+
+			bot.setHtmlWithSelection( 'Foo [<span style="' + ( background ? 'background-' : '' ) + 'color:red">bar</span>]' );
+
+			colorBtn.click( editor );
+
+			setTimeout( function() {
```

Comment:
Timeout with a _magic number_ is not a best way to write code/tests. Isn't there a clear way to hook a proper event that will allow for proper handling of this?